### PR TITLE
Feature/eco 3404 atomic design

### PR DESCRIPTION
**Pull request by DimanKadykov**
- Developer(s): @DimanKadykov

#### Please confirm

- [ ] No new OS components - or they have been approved by legal department

#### Documentation

- [ ] Functional documentation provided or in progress?
- [ ] Integration guide for projects provided or not needed?
- [ ] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   Payolution            | minor                 |                       |

#### Release Notes

{skip}

-----------------------------------------

##### Change log

Improvements

Twig templates update according to the atomic design

### CODE REVIEW

File: src/SprykerEco/Yves/Payolution/Theme/default/installment.twig [hunk 1]
```diff
@@ -1,41 +0,0 @@
-<div class="payolution-form payolution-installment-form payment-form small-12 medium-9 large-6" id="payolutionInstallment">
```
Comment (aleksey-kotsuba) [inline src/SprykerEco/Yves/Payolution/Theme/default/installment.twig]: Do not delete these old files. You can leave them as BC. @tolerants what do you think?

**Comment by aleksey-kotsuba** [comment]
Please provide template for PR.

File: src/SprykerEco/Yves/Payolution/Theme/default/components/molecules/installment/installment.twig [hunk 2]
```diff
@@ -0,0 +1,48 @@
+{% define config = {
+    name: 'installment',
+    tag: 'installment',
```
Comment (tolerants) [inline src/SprykerEco/Yves/Payolution/Theme/default/components/molecules/installment/installment.twig]: Why do you need a custom element here? Suggest to use `div`.

File: src/SprykerEco/Yves/Payolution/Theme/default/components/molecules/installment/installment.twig [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+} %}
+
+{% block body %}
+    <div class="payolution-form payolution-installment-form payment-form small-12 medium-9 large-6" id="payolutionInstallment">
```
Comment (tolerants) [inline src/SprykerEco/Yves/Payolution/Theme/default/components/molecules/installment/installment.twig]: Why we are using foundation grid classes here?

File: src/SprykerEco/Yves/Payolution/Theme/default/components/molecules/invoice/invoice.twig [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{% define config = {
+    name: 'invoice',
+    tag: 'invoice',
```
Comment (tolerants) [inline src/SprykerEco/Yves/Payolution/Theme/default/components/molecules/invoice/invoice.twig]: The same question.

File: src/SprykerEco/Yves/Payolution/Theme/default/components/molecules/installment/installment.twig [hunk 5]
```diff
@@ -0,0 +1,48 @@
+{% define config = {
+    name: 'installment',
+    tag: 'installment',
+} %}
+
```
Comment (tolerants) [inline src/SprykerEco/Yves/Payolution/Theme/default/components/molecules/installment/installment.twig]: Please define a data contract.

File: src/SprykerEco/Yves/Payolution/Theme/default/components/molecules/invoice/invoice.twig [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+} %}
+
```
Comment (tolerants) [inline src/SprykerEco/Yves/Payolution/Theme/default/components/molecules/invoice/invoice.twig]: Please define a data contract.

File: src/SprykerEco/Yves/Payolution/Theme/default/components/molecules/invoice/invoice.twig [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+{% block body %}
+    <div class="payolution-form payolution-invoice-form payment-form small-12 medium-9 large-6" id="payolutionInvoice">
+        <div class="small-12 columns">
+            <label>
+                {{ 'payolutionInvoice.date_of_birth' | trans }}*
+                {{ form_widget(data.form.payolutionInvoice.date_of_birth, { 'attr': {'placeholder': 'payolutionInvoice.date_of_birth' | trans } }) }}
```
Comment (tolerants) [inline src/SprykerEco/Yves/Payolution/Theme/default/components/molecules/invoice/invoice.twig]: What if `data.form.payolutionInvoice` is empty?

File: src/SprykerEco/Yves/Payolution/Theme/default/components/molecules/installment/installment.twig [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+{% block body %}
+    <div class="payolution-form payolution-installment-form payment-form small-12 medium-9 large-6" id="payolutionInstallment">
+        <div class="small-12 columns">
+            <label>
+                {{ 'payolutionInstallment.date_of_birth' | trans }}*
+                {{ form_widget(data.form.payolutionInstallment.date_of_birth, { 'attr': {'placeholder': 'payolutionInstallment.date_of_birth' | trans } }) }}
```
Comment (tolerants) [inline src/SprykerEco/Yves/Payolution/Theme/default/components/molecules/installment/installment.twig]: What if `data.form.payolutionInstallment` is empty?

File: src/SprykerEco/Yves/Payolution/Theme/default/views/installment/installment.twig [hunk 9]
```diff
@@ -0,0 +1,13 @@
+{% extends model('template') %}
+
+{% define data = {
+    form: form,
+} %}
+
+{% block body %}
+    {% include molecule('installment', 'Payolution') with {
+        data: {
+            form: data.form,
```
Comment (tolerants) [inline src/SprykerEco/Yves/Payolution/Theme/default/views/installment/installment.twig]: Maybe it's better to pass `form.payolutionInstallment`?

File: src/SprykerEco/Yves/Payolution/Theme/default/views/invoice/invoice.twig [hunk 10]
```diff
@@ -0,0 +1,13 @@
+{% extends model('template') %}
+
+{% define data = {
+    form: form,
+} %}
+
+{% block body %}
+    {% include molecule('invoice', 'Payolution') with {
+        data: {
+            form: data.form,
```
Comment (tolerants) [inline src/SprykerEco/Yves/Payolution/Theme/default/views/invoice/invoice.twig]: Maybe it's better to pass `form.payolutionInvoice `?

File: src/SprykerEco/Yves/Payolution/Theme/default/components/molecules/installment/installment.twig (hunk 3 above)
Comment (tolerants) [inline src/SprykerEco/Yves/Payolution/Theme/default/components/molecules/installment/installment.twig]: Where all these classes are used? `payolution-form payolution-installment-form payment-for`

File: src/SprykerEco/Yves/Payolution/Theme/default/components/molecules/invoice/invoice.twig [hunk 11]
```diff
@@ -0,0 +1,16 @@
+{% define config = {
+    name: 'invoice',
+    tag: 'invoice',
+} %}
+
+{% block body %}
+    <div class="payolution-form payolution-invoice-form payment-form small-12 medium-9 large-6" id="payolutionInvoice">
```
Comment (tolerants) [inline src/SprykerEco/Yves/Payolution/Theme/default/components/molecules/invoice/invoice.twig]: Where all these classes are used? `payolution-form payolution-invoice-form payment-form`

File: src/SprykerEco/Yves/Payolution/Theme/default/views/installment/installment.twig [hunk 12]
```diff
@@ -0,0 +1,13 @@
+{% extends model('template') %}
+
+{% define data = {
+    form: form,
```
Comment (tolerants) [inline src/SprykerEco/Yves/Payolution/Theme/default/views/installment/installment.twig]: Can we get a `form` as a property of a `_view` object?

File: src/SprykerEco/Yves/Payolution/Theme/default/views/invoice/invoice.twig [hunk 13]
```diff
@@ -0,0 +1,13 @@
+{% extends model('template') %}
+
+{% define data = {
+    form: form,
```
Comment (tolerants) [inline src/SprykerEco/Yves/Payolution/Theme/default/views/invoice/invoice.twig]: Can we get a `form` as a property of a `_view` object?

File: src/SprykerEco/Yves/Payolution/Theme/default/installment.twig (hunk 1 above)
Comment (tolerants) [reply src/SprykerEco/Yves/Payolution/Theme/default/installment.twig]: Agree, but suggest to add a comment, that these files are deprecated.

File: src/SprykerEco/Yves/Payolution/Theme/default/components/molecules/installment/installment.twig [hunk 14]
```diff
@@ -0,0 +1,51 @@
+{% define config = {
+    name: 'installment'
+} %}
+
+{% define data = {
+    form: data.form
```
Comment (tolerants) [inline src/SprykerEco/Yves/Payolution/Theme/default/components/molecules/installment/installment.twig]:
```suggestion
    form: required,
```

File: src/SprykerEco/Yves/Payolution/Theme/default/components/molecules/invoice/invoice.twig [hunk 15]
```diff
@@ -0,0 +1,19 @@
+{% define config = {
+    name: 'invoice'
+} %}
+
+{% define data = {
+    form: data.form
```
Comment (tolerants) [inline src/SprykerEco/Yves/Payolution/Theme/default/components/molecules/invoice/invoice.twig]:
```suggestion
    form: required,
```

File: src/SprykerEco/Yves/Payolution/Theme/default/views/invoice/invoice.twig [hunk 16]
```diff
@@ -0,0 +1,9 @@
+{% extends model('template') %}
```
Comment (tolerants) [inline src/SprykerEco/Yves/Payolution/Theme/default/views/invoice/invoice.twig]:
```suggestion
{% extends model('widget') %}
```

File: src/SprykerEco/Yves/Payolution/Theme/default/views/installment/installment.twig [hunk 17]
```diff
@@ -0,0 +1,9 @@
+{% extends model('template') %}
```
Comment (tolerants) [inline src/SprykerEco/Yves/Payolution/Theme/default/views/installment/installment.twig]:
```suggestion
{% extends model('widget') %}
```

File: src/SprykerEco/Yves/Payolution/Theme/default/views/installment/installment.twig [hunk 18]
```diff
@@ -0,0 +1,9 @@
...
+
+{% block body %}
+    {% include molecule('installment', 'Payolution') with {
+        data: {
+            form: _view.paymentForm.payolutionInstallment,
```
Comment (tolerants) [inline src/SprykerEco/Yves/Payolution/Theme/default/views/installment/installment.twig]: Please add a data contract for this view.

File: src/SprykerEco/Yves/Payolution/Theme/default/views/invoice/invoice.twig [hunk 19]
```diff
@@ -0,0 +1,9 @@
...
+
+{% block body %}
+    {% include molecule('invoice', 'Payolution') with {
+        data: {
+            form: _view.paymentForm.payolutionInvoice,
```
Comment (tolerants) [inline src/SprykerEco/Yves/Payolution/Theme/default/views/invoice/invoice.twig]: Please add a data contract for this view.

File: src/SprykerEco/Yves/Payolution/Theme/default/views/installment/installment.twig [hunk 20]
```diff
@@ -0,0 +1,13 @@
+{% extends template('widget') %}
+
+{% define data = {
+    form: _view.paymentForm.payolutionInstallment,
+} %}
+
+{% block body %}
+    {% include molecule('installment', 'Payolution') with {
+        data: {
+            form: _view.paymentForm.payolutionInstallment,
```
Comment (tolerants) [inline src/SprykerEco/Yves/Payolution/Theme/default/views/installment/installment.twig]:
```suggestion
            form: data.form,
```

File: src/SprykerEco/Yves/Payolution/Theme/default/views/invoice/invoice.twig [hunk 21]
```diff
@@ -0,0 +1,13 @@
+{% extends template('widget') %}
+
+{% define data = {
+    form: _view.paymentForm.payolutionInvoice,
+} %}
+
+{% block body %}
+    {% include molecule('invoice', 'Payolution') with {
+        data: {
+            form: _view.paymentForm.payolutionInvoice,
```
Comment (tolerants) [inline src/SprykerEco/Yves/Payolution/Theme/default/views/invoice/invoice.twig]:
```suggestion
            form: data.form,
```

File: src/SprykerEco/Yves/Payolution/Theme/default/components/molecules/invoice/invoice.twig [hunk 22]
```diff
@@ -0,0 +1,19 @@
+{% define config = {
+    name: 'invoice'
```
Comment (tolerants) [inline src/SprykerEco/Yves/Payolution/Theme/default/components/molecules/invoice/invoice.twig]:
```suggestion
    name: 'invoice',
```

File: src/SprykerEco/Yves/Payolution/Theme/default/components/molecules/installment/installment.twig [hunk 23]
```diff
@@ -0,0 +1,51 @@
+{% define config = {
+    name: 'installment'
```
Comment (tolerants) [inline src/SprykerEco/Yves/Payolution/Theme/default/components/molecules/installment/installment.twig]:
```suggestion
    name: 'installment',
```

**Pull request merged**